### ISSUE TITLE
[kube-prometheus-stack] Enable vertical pod autoscaler support for prometheus

### ISSUE THREAD

**Issue by z-anshun** [prometheus-community/helm-charts#3095]
### Is your feature request related to a problem ?

Currently, the chart enables `VPA` support for `prometheus-operator`,and I want `prometheus` to support `VPA` too.

There are two reasons:
1. In many cases, we will not or have not noticed that the resources of the default `prometheus` need to be adjusted. 
2.The current VPA already allows to change container resource requests & limits without necessarily restarting the container.

### Describe the solution you'd like.

Support vertical pod autoscaler just like `prometheus-operator`.

/kube-prometheus-stack/templates/prometheus/verticalpodautoscaler.yaml:
```yaml
{{- if and (.Capabilities.APIVersions.Has "autoscaling.k8s.io/v1") (.Values.prometheus.verticalPodAutoscaler.enabled) }}
apiVersion: autoscaling.k8s.io/v1
kind: VerticalPodAutoscaler
metadata:
  name: {{ template "kube-prometheus-stack.fullname" . }}-prometheus
  namespace: {{ template "kube-prometheus-stack.namespace" . }}
  labels:
    app: {{ template "kube-prometheus-stack.name" . }}-prometheus
{{ include "kube-prometheus-stack.labels" . | indent 4 }}
spec:
  resourcePolicy:
    containerPolicies:
      - containerName: "prometheus"
      {{- if .Values.prometheus.verticalPodAutoscaler.controlledResources }}
        controlledResources: {{ .Values.prometheus.verticalPodAutoscaler.controlledResources }}
      {{- end }}
      {{- if .Values.prometheus.verticalPodAutoscaler.maxAllowed }}
        maxAllowed:
        {{- toYaml .Values.prometheus.verticalPodAutoscaler.maxAllowed | nindent 8 }}
      {{- end }}
      {{- if .Values.prometheus.verticalPodAutoscaler.minAllowed }}
        minAllowed:
        {{- toYaml .Values.prometheus.verticalPodAutoscaler.minAllowed | nindent 8 }}
      {{- end }}
  targetRef:
    apiVersion: apps/v1
    kind: StatefulSet
    name: prometheus-{{ include "kube-prometheus-stack.prometheus.crname" $ }}
  {{- if .Values.prometheus.verticalPodAutoscaler.updatePolicy }}
  updatePolicy:
    {{- if .Values.prometheus.verticalPodAutoscaler.updatePolicy.updateMode }}
    updateMode: {{ .Values.prometheus.verticalPodAutoscaler.updatePolicy.updateMode  }}
    {{- end }}
  {{- end }}
{{- end }}
```

### Describe alternatives you've considered.

`NONE`

**Comment by atsai1220** [comment]
This would be a boon when managing multiple clusters with varying degrees of resource requirements.

I've tried omitting resource requests and limits inside the `Prometheus` CRD instance then update the resultant `statefulset` Prometheus. The operator will remove the additional resource requests/limits I configured out-of-band from the CRD (emulating VPA behavior).

I would almost expect fields not declared inside the instance of the Prometheus CRD would allow edit capability. Configuring the operator to ignore resource requests if it's not set will allow VPA to freely manage the Prometheus statefulset.
